torch/csrc/jit/graph_executor.cpp

```diff
@@ -708,6 +711,11 @@ void runNondiffOptimization(std::shared_ptr<Graph>& graph) {
   // Rewrite subgraphs with many MMs into expressions that batch them.
   BatchMM(graph);
 
+  if (at::globalContext().qEngine() == at::kFBGEMM ||
+      at::globalContext().qEngine() == at::kQNNPACK) {
```

Comment:
Optimization pipeline in graph executor cannot be conditionalized on a global state. We need to make sure QuantFusion is always correct and run it unconditionally.